File: app/src/main/java/com/peruukki/simplecurrencyconverter/db/ConverterContract.java
package com.peruukki.simplecurrencyconverter.db;

import android.net.Uri;
import android.provider.BaseColumns;

public abstract class ConverterContract {

    private static final String CONTENT_AUTHORITY = "com.peruukki.simplecurrencyconverter";

    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + CONTENT_AUTHORITY);

    public static final class ConversionRateEntry implements BaseColumns {
        public static final String TABLE_NAME = "conversionRates";

        public static final String COLUMN_FIXED_CURRENCY = "fixed";
        public static final String COLUMN_VARIABLE_CURRENCY = "variable";
        public static final String COLUMN_CONVERSION_RATE = "rate";
    }
}

File: app/src/main/java/com/peruukki/simplecurrencyconverter/db/ConverterContract.kt
package com.peruukki.simplecurrencyconverter.db

import android.net.Uri
import android.provider.BaseColumns

object ConverterContract {

    private val CONTENT_AUTHORITY = "com.peruukki.simplecurrencyconverter"

    @JvmField
    val BASE_CONTENT_URI = Uri.parse("content://$CONTENT_AUTHORITY")

    class ConversionRateEntry : BaseColumns {
        companion object {
            const val TABLE_NAME = "conversionRates"

            const val COLUMN_FIXED_CURRENCY = "fixed"
            const val COLUMN_VARIABLE_CURRENCY = "variable"
            const val COLUMN_CONVERSION_RATE = "rate"
        }
    }
}
